src/dmd/dtoh.d

```diff
@@ -966,100 +1034,165 @@ public:
             //return;
         //}
 
-        bool hasBaseType = false;
-
-        switch (ed.memtype.ty)
+        // we need to know a bunch of stuff about the enum...
+        bool isAnonymous = ed.ident is null;
+        AST.Type type = ed.memtype;
+        if (!type)
         {
-            case AST.Tbool, AST.Tvoid:
-            case AST.Tchar, AST.Twchar, AST.Tdchar:
-            case AST.Tint8, AST.Tuns8:
-            case AST.Tint16, AST.Tuns16:
-            case AST.Tint64, AST.Tuns64:
-            case AST.Tfloat32, AST.Tfloat64, AST.Tfloat80:
-                hasBaseType = true;
-                break;
-            case AST.Tint32, AST.Tuns32, AST.Tenum: // by default, the base is an int
-                break;
-            default:
-                import dmd.root.string : toDString;
-                printf ("%s\n", ed.ident.toChars());
-                assert(0, ed.memtype.kind.toDString);
+            // check all keys have matching type
+            foreach (i, _m; *ed.members)
+            {
+                AST.EnumMember m = cast(AST.EnumMember)_m;
```

Comment:
Using:
```
auto m = _m.isEnumMember();
```
avoids a dangerous cast.

Reply:
Okay. But how could it be possible that `members` in `EnumDeclaration` are not `EnumMember`s